test/check-machines-consoles

```diff
@@ -324,6 +325,124 @@ fullscreen=0
 
         self.waitViewerDownload("vnc", my_ip)
 
+    def testAddEditVNC(self, withSerial=True):
+        b = self.browser
+
+        # Create a machine without any consoles
+
+        name = "subVmTest1"
+        self.createVm(name, ptyconsole=withSerial)
+
+        self.login_and_go("/machines")
+        self.waitPageInit()
+        self.waitVmRow(name)
+        self.goToVmPage(name)
+
+        def assert_state(text):
+            b.wait_in_text(f"#vm-{name}-consoles .pf-v5-c-empty-state", text)
+
+        def assert_not_state(text):
+            b.wait_not_in_text(f"#vm-{name}-consoles .pf-v5-c-empty-state", text)
+
+        if withSerial:
+            b.click("#pf-v5-c-console__type-selector")
+            b.wait_visible("#pf-v5-c-console__type-selector + .pf-v5-c-select__menu")
+            b.click("#VncConsole button")
+            b.wait_not_present("#pf-v5-c-console__type-selector + .pf-v5-c-select__menu")
+
+        # "Console" card shows empty state
+
+        if withSerial:
+            assert_state("Graphical console support not enabled")
+            b.assert_pixels(f"#vm-{name}-consoles", "no-vnc")
+        else:
+            assert_state("Console support not enabled")
+
+        b.click(f"#vm-{name}-consoles .pf-v5-c-empty-state button:contains(Add VNC)")
+
+        b.wait_visible("#add-vnc-dialog")
+        b.set_input_text("#add-vnc-port", "5000")
+        b.click("#add-vnc-save")
+        b.wait_visible("#add-vnc-dialog .pf-m-error:contains('Port must be 5900 or larger.')")
+        b.set_input_text("#add-vnc-port", "Newport, Rhode Island")
```

Comment:
Hamburg! (Thanks for the chuckle :laughing: )

Reply:
Yes, I'll change this to Hamburg! (Got family there even.)